app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,337 @@
+        isError = currentUser is ErrorResult || currentUserStats is ErrorResult || currentUserProjects is ErrorResult
+    )
+}
+
+@Composable
+fun ProfileScreenContentTest(
+    navigateBack: () -> Unit = {},
+    currentUser: User? = null,
+    currentUserStats: Stats? = null,
+    currentUserProjects: List<Project> = emptyList(),
+    isLoading: Boolean = false,
+    isError: Boolean = false
+) = Column(
+    modifier = Modifier.fillMaxSize(),
+    horizontalAlignment = Alignment.CenterHorizontally
+) {
+    AppBarWithBackButton(
+        title = { Text(stringResource(R.string.profile)) },
+        navigateBack = navigateBack
+    )
+
+    when {
+        isLoading || isError -> {
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.Center
+            ) {
+                CircularLoader()
+            }
+        }
+        else -> {
+            ProfileScreenContent(
+                currentUser = currentUser,
+                currentUserStats = currentUserStats,
+                currentUserProjects = currentUserProjects,
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalCoilApi::class)
+@Composable
+fun ProfileScreenContent(
+    currentUser: User? = null,
+    currentUserStats: Stats? = null,
+    currentUserProjects: List<Project> = emptyList()
+) {
+    Image(
+        painter = rememberImagePainter(
+            data = currentUser?.avatarUrl ?: R.drawable.default_avatar,
+            builder = {
+                error(R.drawable.default_avatar)
+                crossfade(true)
+            },
+        ),
+        contentDescription = null,
+        contentScale = ContentScale.Crop,
+        modifier = Modifier
+            .size(200.dp)
+            .clip(CircleShape)
+    )
+
+    Spacer(Modifier.height(16.dp))
+
+    Text(
+        text = currentUser?.fullName ?: "Full name",
+        style = MaterialTheme.typography.titleLarge
+    )
+
+    Spacer(Modifier.height(2.dp))
+
+    Text(
+        text = "@${currentUser?.username ?: "username"}",
+        color = MaterialTheme.colorScheme.outline,
+        style = MaterialTheme.typography.bodyLarge,
+    )
+
+    Spacer(Modifier.height(24.dp))
+
+    Row(
+        horizontalArrangement = Arrangement.SpaceEvenly,
+        modifier = Modifier.fillMaxWidth()
+    ) {
+        ColumnTextData(currentUserStats?.totalNumProjects.toString(), stringResource(R.string.projects))
+        ColumnTextData(
+            currentUserStats?.totalNumClosedUserStories.toString(),
+            stringResource(R.string.closed_user_story)
+        )
+        ColumnTextData(currentUserStats?.totalNumContacts.toString(), stringResource(R.string.contacts))
+    }
+
+    Spacer(Modifier.height(24.dp))
+
+    Text(
+        text = stringResource(R.string.projects),
+        style = MaterialTheme.typography.titleLarge
+    )
+
+    Spacer(Modifier.height(12.dp))
+
+    LazyColumn(Modifier.padding(horizontal = mainHorizontalScreenPadding)) {
```

Comment:
It's better to put everything inside this `LazyColumn`. For now, only projects list is scrollable, which looks strange and not very user-friendly